wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift

```diff
@@ -185,86 +195,77 @@ final class CreateGroupConversationActionHandler: ActionHandler<CreateGroupConve
         _ response: ZMTransportResponse,
         action: CreateGroupConversationAction
     ) {
-        var action = action
-
-        guard
-            let apiVersion = APIVersion(rawValue: response.apiVersion),
-            let rawData = response.rawData,
-            let payload = Payload.Conversation(rawData, apiVersion: apiVersion),
-            let newConversation = processor.updateOrCreateConversation(
-                from: payload,
-                in: context
-            )
-        else {
-            Logging.network.warn("Can't process response, aborting.")
-            action.fail(with: .proccessingError)
-            return
-        }
+        Task {
```

Comment:
I thought we were going to turn this async directly? But i supposed we only need it in this action handler, right?

What about cancelling the task if we're cancelling/removing the action

Reply:
How can I turn it async directly? I'm all for it if it's possible.

> What about cancelling the task if we're cancelling/removing the action

What do you mean? This is the function which handles the response from the backend. How can it be cancelled?

Reply:
Ok, yeah that's nicer